src/Effect/Logger.hs

```diff
@@ -134,6 +141,33 @@ withWriterLogger sev = runLogger ctx
         , logCtxWrite = tell . pure @f
         }
 
+-- | Like `withWriterLogger`, except it works even when the logged action forks
+-- threads.
+withTVarLogger ::
+  forall f sig m a.
+  ( Applicative f
+  , Monoid (f (Doc AnsiStyle))
+  , Has (Reader (TVar (f (Doc AnsiStyle)))) sig m
```

Comment:
Why are we using a mutable variable instead of a queue?  Having a single variable means lock contention.

Reply:
Maybe add some documentation on what you discovered so that future engineers will feel empowered to make the change if and when it makes sense to them.